### PR TITLE
Applied new structure and lazy route

### PR DESCRIPTION
**Pull request by FlavioShift3**
I did apply changes to keep similar with [https://github.com/Shift3/boilerplate-client-angular](https://github.com/Shift3/boilerplate-client-angular
)
- Created new folders
- Move components to its respective folder
- Modified route to lazy route
- Fix Unit Test (Karma, Mocha, and Jasmine) to work
- Replace ListView to RadListView, Listview have issues with lazy route.

### CODE REVIEW

File: README copy.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# SP-NativeScript
```
Comment (Basim-Elayan) [inline README copy.md:1]: Should be renamed to readme.md

Comment (FlavioShift3) [reply README copy.md]: Done

File: src/app/app-routing.module.ts [hunk 2]
```diff
@@ -1,16 +1,21 @@
...
-    { path: "login", component: LoginComponent },
-    { path: "items", component: ItemsComponent },
-    { path: "item/:id", component: ItemDetailComponent }
+    {
+        path: '',
+        redirectTo: 'auth',
+        pathMatch: 'full',
+    },
+    {
+        path: 'auth',
```
Comment (Basim-Elayan) [inline src/app/app-routing.module.ts:12]: Awesome, it's really nice to start using Lazy Loading with NS.

File: webpack.config.js [hunk 3]
```diff
@@ -182,6 +182,7 @@ module.exports = env => {
             "setImmediate": false,
             "fs": "empty",
             "__dirname": false,
+            "child_process": "empty",
```
Comment (Basim-Elayan) [inline webpack.config.js:185]: What is this for?

File: src/app/app.module.ts [hunk 4]
```diff
@@ -24,9 +21,6 @@ import { ReleaseService } from "./services/release.service";
     ],
     declarations: [
         AppComponent,
-        ItemsComponent,
```
Comment (Basim-Elayan) [inline src/app/app.module.ts:27]: Much better and organized more.

File: src/app/infrastructure/services/release.service.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class ReleaseService {
```
Comment (Basim-Elayan) [inline src/app/infrastructure/services/release.service.ts:6]: I wonder what the purpose of is this services exactly?

File: webpack.config.js (hunk 3 above)
Comment (FlavioShift3) [reply webpack.config.js:185]: I saw this on readme file. That way I add that here.

File: src/app/infrastructure/services/release.service.ts (hunk 5 above)
Comment (FlavioShift3) [reply src/app/infrastructure/services/release.service.ts:6]: To make this service a singleton. When we declare on component. That make sure this service is a singleton.

**Pull request merged**